api/player.go

```diff
@@ -8,15 +8,16 @@ import (
 	"strings"
 
 	"github.com/davidlu1997/gogogo/model"
+	"github.com/gorilla/mux"
 )
 
 func InitPlayer() {
 	BaseRoutes.Players.Handle("/create", ApiHandler(createPlayer)).Methods("POST")
-
 	BaseRoutes.NeedPlayer.Handle("/update", ApiPlayerRequired(updatePlayer)).Methods("POST")
 	BaseRoutes.NeedPlayer.Handle("/get", ApiPlayerRequired(getPlayer)).Methods("GET")
 	BaseRoutes.NeedPlayer.Handle("/games", ApiPlayerRequired(getPlayerGames)).Methods("GET")
-	BaseRoutes.NeedPlayer.Handle("/find", ApiPlayerRequired(findPlayer)).Methods("POST")
+	BaseRoutes.NeedPlayer.Handle("/get_email/{email:[A-Za-z0-9+.@]}", ApiPlayerRequired(getPlayerByEmail)).Methods("GET")
```

Comment:
No, I think `_` and `-` are valid characters.

Source: https://stackoverflow.com/questions/1547899/which-characters-make-a-url-invalid